tests/test_arcaflow_plugin_pcp.py

```diff
@@ -113,77 +113,44 @@ def test_serialization():
-        )
-
-        print(f"==>> output_id is {output_id}")
-        print(f"==>> output_data is {output_data}")
-
-        self.assertEqual("success", output_id)
-        plugin.test_object_serialization(pcp_plugin.PerfOutput(output_data.pcp_output))
-
-    def test_functional_flat(self):
-        input = pcp_plugin.PcpInputParams(
-            pmlogger_interval=1.0,
-            pmlogger_metrics="kernel.all.cpu.user mem.util.used",
-            timeout=3,
-            flatten=True,
-        )
-
-        output_id, output_data = pcp_plugin.StartPcpStep.start_pcp(
-            params=input, run_id="ci_pcp"
-        )
-
-        print(f"==>> output_id is {output_id}")
-        print(f"==>> output_data is {output_data}")
-
-        self.assertEqual("success", output_id)
-        plugin.test_object_serialization(pcp_plugin.PerfOutput(output_data.pcp_output))
-
-    def test_functional_user_pmlogger_conf(self):
-        pmlogger_conf = Path("tests/pmlogger.conf").read_text()
-        input = pcp_plugin.PcpInputParams(
-            pmlogger_interval=1.0,
-            pmlogger_metrics="kernel.all.cpu.user mem.util.used",
-            timeout=3,
-            pmlogger_conf=pmlogger_conf,
-        )
-
-        output_id, output_data = pcp_plugin.StartPcpStep.start_pcp(
-            params=input, run_id="ci_pcp"
-        )
-
-        print(f"==>> output_id is {output_id}")
-        print(f"==>> output_data is {output_data}")
-
-        self.assertEqual("success", output_id)
-        plugin.test_object_serialization(pcp_plugin.PerfOutput(output_data.pcp_output))
-
-    def test_functional_user_pmrep_conf(self):
-        pmrep_conf = Path("tests/pmrep.conf").read_text()
-        input = pcp_plugin.PcpInputParams(
-            pmlogger_interval=1.0,
-            pmlogger_metrics="kernel.all.cpu.user mem.util.used",
-            timeout=3,
-            pmrep_conf=pmrep_conf,
-        )
-
-        output_id, output_data = pcp_plugin.StartPcpStep.start_pcp(
-            params=input, run_id="ci_pcp"
-        )
+        tests = [
+            {
+                # Standard
+            },
+            {
+                # Flatten
+                "flatten": True
+            },
+            {
+                # User-provided pmlogger config
+                "pmlogger_conf": Path("tests/pmlogger.conf").read_text()
+            },
+            {
+                # User-provided pmrep config
+                "pmrep_conf": Path("tests/pmrep.conf").read_text()
+            },
+        ]
+
+        for test in tests:
+            with self.subTest(test=test):
+                input = pcp_plugin.PcpInputParams(
+                    pmlogger_interval=1.0,
+                    pmlogger_metrics="kernel.all.cpu.user mem.util.used",
+                    timeout=3,
+                    **test,
+                )
+
+            output_id, output_data = pcp_plugin.StartPcpStep.start_pcp(
+                params=input, run_id="ci_pcp"
+            )
```

Comment:
I think we want lines 143-145 indented one more level, so that they are executed with each test/input.

And, presumably, the same applies to line 147-153.